mmv1/third_party/terraform/services/osconfigv2/resource_os_config_v2_policy_orchestrator_for_folder_test.go.tmpl

```diff
@@ -0,0 +1,286 @@
+package osconfigv2_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccOSConfigV2PolicyOrchestratorForFolder_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org_id":        envvar.GetTestOrgFromEnv(t),
+		"zone":          envvar.GetTestZoneFromEnv(),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccOSConfigV2PolicyOrchestratorForFolder_basic(context),
+			},
+			{
+				ResourceName:            "google_os_config_v2_policy_orchestrator_for_folder.policy_orchestrator_for_folder",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"folder_id", "labels", "policy_orchestrator_id", "terraform_labels"},
+			},
+			{
+				Config: testAccOSConfigV2PolicyOrchestratorForFolder_update(context),
```

Comment:
This needs a ConfigPlanCheck to make sure the resource is actually being updated in-place (not deleted and recreated). I don't anticipate it being a problem at the moment since most of the fields are not immutable, but it would be good to have the check in place.

```suggestion
				Config: testAccOSConfigV2PolicyOrchestratorForFolder_update(context),
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("google_os_config_v2_policy_orchestrator_for_folder.policy_orchestrator_for_folder", plancheck.ResourceActionUpdate),
					},
				},
```